BetaUtils.Core/Exceptions/Middleware/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+
+namespace BetaUtils.Core.Exceptions.Middleware;
```

Comment:
I am not sure this is the right place to put middleware stuff.  I think this for two reasons

1. The concept of middleware highly relies on http context, therefore it probably should not be a part of core. For example, this is not something you would want to have in your application if you are making desktop app or class library. Rather than this, we could create a new project "BetaUtils.Api.Core" which could contain things that are intended only for the API projects, where we actually need them. 
2. In the folder called "Exceptions" we should expect to see only exceptions, but not the complex classes such are middlewares and their support classes. Rather than that we could create a folder called "Middlewares" where we could create this, and many more middlewares.

Reply:
I am so happy to see this question popping up ❤️. Yes BetaUtils.Api.Core is more suitable for this so I will be moving it there, as of why I am happy about question...

I personally wouldn't place it anywhere here, this type of middleware / handling would be perfect for `package` what bring us to topic of packages / services.
Structures / codes that should be in package from beginning we should place there and services should be set up with "bare minimum logic" so each addition to code / packages could be shown on example.
Taking for example this PR it would take some amount of research and debugging to implement this logic since it differs depending on env (.net core, .net framework, .net6...).

As first aid I will move it to BetaUtils.Api.Core.